### ISSUE TITLE
React: false continues to listen for scale updates

### ISSUE THREAD

**Issue by arvind** [vega/vega#1428]
The following spec causes an infinite loop `signal(box_x)` -> `signal(box_Horsepower)` -> `signal(box_tuple)` -> `signal(box_modify)` -> `signal(box_store)` -> `filter(data_0)` -> `scale(x)` -> `signal(box_x)`. This is surprising as although `box_x` references `scale('x')` within its `update` function, it is marked as `react: false`. 

```json
{
  "$schema": "https://vega.github.io/schema/vega/v4.json",
  "autosize": "pad",
  "padding": 5,
  "width": 200,
  "height": 200,
  "data": [
    {"name": "box_store"},
    {
      "name": "source_0",
      "url": "data/cars.json"
    },
    {
      "name": "data_0",
      "source": "source_0",
      "transform": [
        {
          "type": "filter",
          "expr": "!(length(data(\"box_store\"))) || (vlSelectionTest(\"box_store\", datum))"
        }
      ]
    }
  ],
  "signals": [
    {
      "name": "unit",
      "value": {},
      "on": [
        {"events": "mousemove", "update": "isTuple(group()) ? group() : unit"}
      ]
    },
    {"name": "box", "update": "vlSelectionResolve(\"box_store\")"},
    {
      "name": "box_x",
      "update": "warn('box_x', [scale(\"x\", 100),scale(\"x\", 150)])",
      "react": false
    },
    {
      "name": "box_Horsepower",
      "value": [100, 150],
      "on": [
        {
          "events": {"signal": "box_x"},
          "update": "warn('box_Horsepower_on', box_x[0] === box_x[1] ? null : invert(\"x\", box_x))"
        }
      ]
    },
    {
      "name": "box_tuple",
      "update": "warn('box_tuple_init', {unit: \"layer_0\", fields: box_tuple_fields, values: [[100,150]]})",
      "react": false,
      "on": [
        {
          "events": [{"signal": "box_Horsepower"}],
          "update": "warn('box_tuple_on', box_Horsepower ? {unit: \"layer_0\", fields: box_tuple_fields, values: [box_Horsepower]} : null)"
        }
      ]
    },
    {
      "name": "box_tuple_fields",
      "update": "[{\"field\":\"Horsepower\",\"channel\":\"x\",\"type\":\"R\"}]"
    },
    {"name": "box_modify", "update": "warn('box_modify', modify(\"box_store\", box_tuple, true))"}
  ],
  "scales": [
    {
      "name": "x",
      "type": "linear",
      "domain": {
        "fields": [
          {"data": "source_0", "field": "Horsepower"},
          {"data": "data_0", "field": "Horsepower"}
        ]
      },
      "range": [0, {"signal": "width"}],
      "nice": true,
      "zero": true
    }
  ]
}
```

**Comment by jheer** [comment]
Thank you! This is indeed a bug. I located the problem in the vega-runtime module: operator instantiation and parameter initialization are performed in separate passes. The parameter initialization stage erroneously omits the react flag here: https://github.com/vega/vega-runtime/blob/master/src/operator.js#L28

I will this fix promptly and include in the upcoming release!

**Comment by jheer** [comment]
Fixed in vega version 4.3.0.
